ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape_pybind.cpp

```diff
@@ -22,22 +24,45 @@ void bind_reshape_view(pybind11::module& module, const data_movement_operation_t
         operation,
         doc,
         ttnn::pybind_overload_t{
-            [](const data_movement_operation_t& self, const ttnn::Tensor& input_tensor, const ttnn::Shape& shape)
-                -> ttnn::Tensor { return self(input_tensor, shape); },
+            [](const data_movement_operation_t& self,
+               const ttnn::Tensor& input_tensor,
+               const ttnn::Shape& shape,
+               const std::optional<MemoryConfig> &memory_config,
+                const uint8_t queue_id,
+                const std::optional<PadValue> &pad_value
+               ) -> ttnn::Tensor {
+                return self(input_tensor, shape);
+            },
             py::arg("input_tensor"),
             py::arg("shape"),
-        },
+            py::kw_only(),
+            py::arg("memory_config") = std::nullopt,
+            py::arg("queue_id") = 0,
+            py::arg("pad_value") = std::nullopt
+            },
         ttnn::pybind_overload_t{
             [](const data_movement_operation_t& self,
                const ttnn::Tensor& input_tensor,
-               const ttnn::SmallVector<int32_t> shape) -> ttnn::Tensor { return self(input_tensor, shape); },
+               const ttnn::SmallVector<int32_t> shape,
+               const std::optional<MemoryConfig> &memory_config,
```

Comment:
Amazing! Is therr a condition for 0 cost view that this is left as null, because in the case where we want to go from L1 to dram or vice versa or sharded to interleaved or vice versa it will require a data movement

Reply:
Oh true, just added it